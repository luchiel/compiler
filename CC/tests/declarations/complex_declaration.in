int c[20], *b[5 * 6 == 30 ? 4 : 3];