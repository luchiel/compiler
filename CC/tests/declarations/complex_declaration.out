type <int>
type <float>
type <void>
type <int*>
	pointer to <int>
type <void*>
	pointer to <void>
variable NULL of type <void*> initialized by
	{int 0}
		type <int>
function printf
	returns type <int>
	arguments
		<no args>
type <0>
	array [
		{int 20}
			type <int>
	] of <int>
variable c of type <0>
type <1>
	array [
		{?:}
			type <int>
		| 
		+-{==}
				type <int>
		| | 
		| +-{*}
					type <int>
		| | | 
		| | +-{int 5}
						type <int>
		| | | 
		| | +-{int 6}
						type <int>
		| | 
		| +-{int 30}
					type <int>
		| 
		+-{int 4}
				type <int>
		| 
		+-{int 3}
				type <int>
	] of pointer to <int>
variable b of type <1>
